dlog/commitment/src/srs.rs

```diff
@@ -22,6 +22,40 @@ pub struct SRS<G: CommitmentCurve> {
     pub endo_q: G::BaseField,
 }
 
+pub struct TrimmedSRS<'a, G: CommitmentCurve> {
+    pub srs: &'a SRS<G>,
+    pub trimmed_length_log2: usize
+}
+
+impl<G: CommitmentCurve> SRS<G> {
+    pub fn trim<'a>(&'a self, trimmed_length_log2: usize) -> TrimmedSRS<'a, G> {
+        assert!((1 << trimmed_length_log2) <= self.g.len());
+        TrimmedSRS { srs: self, trimmed_length_log2 }
+    }
+}
```

Comment:
it'd be good to have a test to exercise the new functionality

Reply:
Happy to help with this if you want to suggest the test details.

Reply:
I edited the dlog/tests/batch.rs test so that it uses trimming